src/services/rgbpp.ts

```diff
@@ -298,9 +306,87 @@ export default class RgbppCollector extends BaseQueueWorker<IRgbppCollectRequest
     return null;
   }
 
+  async isIsomorphicTx(
+    btcTx: Transaction,
+    ckbTx: CKBComponents.RawTransaction,
+    validateCommitment?: boolean,
+  ): Promise<boolean> {
+    // Find the commitment from the btc_tx
+    const btcTxCommitment = tryGetCommitmentFromBtcTx(btcTx);
+    if (!btcTxCommitment) {
+      return false;
+    }
+
+    // Check inputs:
+    // 1. Find the last index of the type inputs
+    // 2. Check if all rgbpp_lock inputs can be found in the btc_tx.vin
+    // 3. Check if the inputs contain at least one rgbpp_lock cell (as L1-L1 and L1-L2 transactions should have)
+    const inputs = await this.cradle.ckb.getInputCellsByOutPoint(ckbTx.inputs.map((input) => input.previousOutput!));
+    const lastTypeInputIndex = findLastIndex(inputs, (input) => !!input.cellOutput.type);
+    const anyRgbppLockInput = inputs.some((input) => isRgbppLock(input.cellOutput.lock));
+    if (!anyRgbppLockInput) {
+      return false;
+    }
+    const allInputsValid = inputs.every((input, index) => {
+      if (!input.cellOutput.type) {
+        return true;
+      }
+      if (!isRgbppLock(input.cellOutput.lock)) {
+        return true;
+      }
+      const btcInput = btcTx.vin[index];
+      const rgbppLockArgs = unpackRgbppLockArgs(input.cellOutput.lock.args);
+      return btcInput && btcInput.txid === remove0x(rgbppLockArgs.btcTxid) && btcInput.vout === rgbppLockArgs.outIndex;
```

Comment:
This line of code assumes a one-to-one correspondence between CKB transaction inputs and Bitcoin transaction inputs, which may not always be the case.

See https://github.com/ckb-cell/rgbpp/blob/4abd216dbd5d25b8f59e843de39643f9b7513f9d/contracts/rgbpp-lock/src/main.rs#L162-L168

A more robust approach would be to search for a matching Bitcoin input across all btcTx.vin entries, rather than assuming they're at the same index. This would allow for flexibility in input ordering between the two chains.

e.g.
```ts
const matchingBtcInput = btcTx.vin.find(vinput => 
  ...
);
return !!matchingBtcInput;
```

Reply:
Fixed at 808b383